staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -469,7 +469,16 @@ func (r Request) finalURLTemplate() url.URL {
 	groupIndex := 0
 	index := 0
 	if r.URL() != nil && r.baseURL != nil && strings.Contains(r.URL().Path, r.baseURL.Path) {
-		groupIndex += len(strings.Split(r.baseURL.Path, "/"))
+		// only add baseURL path length if it is different than `/`
+		// as strings.Split("/", "/") returns length of 2
+		if r.baseURL.Path != "/" {
+			// We need to perform a `path.Join` as path.Join appends `/` to a
+			// path that does not contain `/` and removes duplicate `/` if they
+			// exist.
+			groupIndex += len(strings.Split(path.Join("/", r.baseURL.Path), "/"))
```

Comment:
Do you need to do the path.Join at line 468 as well?

Reply:
I've added now.

Reply:
nit: s/prepend/prepends
nit: move the comment to line 468?